lib/detect-testing-library-utils.ts

```diff
@@ -13,48 +18,84 @@ export function detectTestingLibraryUtils<
 
-    // TODO: init here options based on shared ESLint config
+    // Init options based on shared ESLint settings
+    const customModule = context.settings['testing-library/module'];
 
-    // helpers for Testing Library detection
+    // Helpers for Testing Library detection.
     const helpers: DetectionHelpers = {
-      getIsImportingTestingLibrary() {
-        return isImportingTestingLibrary;
+      getIsTestingLibraryImported() {
+        if (!customModule) {
+          return true;
+        }
+
+        return isImportingTestingLibraryModule || isImportingCustomModule;
+      },
+      canReportErrors() {
+        return this.getIsTestingLibraryImported();
       },
     };
 
-    // instructions for Testing Library detection
+    // Instructions for Testing Library detection.
+    // `ImportDeclaration` must be first in order to know if Testing Library
+    // is imported ASAP.
     const detectionInstructions: TSESLint.RuleListener = {
       ImportDeclaration(node: TSESTree.ImportDeclaration) {
-        isImportingTestingLibrary = /testing-library/g.test(
-          node.source.value as string
-        );
+        if (!isImportingTestingLibraryModule) {
+          // check only if testing library import not found yet so we avoid
+          // to override isImportingTestingLibraryModule after it's found
+          isImportingTestingLibraryModule = /testing-library/g.test(
+            node.source.value as string
+          );
+        }
+
+        if (!isImportingCustomModule) {
+          // check only if custom module import not found yet so we avoid
+          // to override isImportingCustomModule after it's found
+          const importName = String(node.source.value);
+          isImportingCustomModule = importName.endsWith(customModule);
+        }
       },
     };
 
     // update given rule to inject Testing Library detection
     const ruleInstructions = ruleCreate(context, optionsWithDefault, helpers);
-    const enhancedRuleInstructions = Object.assign({}, ruleInstructions);
+    const enhancedRuleInstructions: TSESLint.RuleListener = {};
+
+    // The order here is important too: detection instructions must come before
+    // than rule instructions to:
+    //  - detect Testing Library things before the rule is applied
+    //  - be able to prevent the rule about to be applied if necessary
+    const allKeys = new Set(
+      Object.keys(detectionInstructions).concat(Object.keys(ruleInstructions))
+    );
 
-    Object.keys(detectionInstructions).forEach((instruction) => {
-      (enhancedRuleInstructions as TSESLint.RuleListener)[instruction] = (
-        node
-      ) => {
+    allKeys.forEach((instruction) => {
+      enhancedRuleInstructions[instruction] = (node) => {
         if (instruction in detectionInstructions) {
           detectionInstructions[instruction](node);
         }
 
-        if (ruleInstructions[instruction]) {
+        if (helpers.canReportErrors() && ruleInstructions[instruction]) {
```

Comment:
Here is where the plugin prevents original rule instructions to be executed if conditions are not met!

Reply:
I'm a bit lost what's the difference between the method `canReportErrors` and `getIsTestingLibraryImported` - given the first one just calls the second

Reply:
in addition to that, if I understand correctly, we would just execute the rules if TL is imported, otherwise, nothing gets executed - is that correct?

Reply:
I'm anticipating next PR with `canReportErrors` so it warps all necessary checks to know if rules should report errors or not. That's just knowing if Testing Library is imported from now, but in the next PR I'll add the shared option to customize the file names pattern which should be checked by regexp, so the user can set to just report from this plugin on files named `*.test.js`. This is interesting when you have different kind of tests files (.test.js for Testing Library and .spec.js for Cypress) or just to reduce the scope of the plugin. So in the next PR `canReportErrors` will check more things and will be in charge of wrap all necessary checks in the future if others needed.

> in addition to that, if I understand correctly, we would just execute the rules if TL is imported, otherwise, nothing gets executed - is that correct?

Yes, but by default TL is always considered as imported due to the "aggressive reporting" mode.